### PR TITLE
Add new post-submit job to build keda-module

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**

Changes proposed in this pull request:

- Add new post-submit job to build keda-module

**Related issue(s)**

See also:
- https://github.com/kyma-project/keda-manager/issues/58
- https://github.com/kyma-project/keda-manager/pull/62

### CODE REVIEW

File: prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml [hunk 1]
```diff
@@ -167,6 +168,47 @@ presubmits: # runs on PRs
   
 postsubmits: # runs on main
   kyma-project/keda-manager:
+    - name: post-make-module-build
```
[the remark this answers is not in the document]
Comment (m00g3n) [reply prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml]: BTW we need to change this in `template/data` and regenerate jobs

File: prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml [hunk 2]
```diff
@@ -167,6 +168,47 @@ presubmits: # runs on PRs
...
+      annotations:
+        pipeline.trigger: "pr-merge"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "post-make-module-build"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-sa-kyma-push-images: "true"
+      run_if_changed: '^(go.mod|go.sum)$|^*/(.*.go|Makefile|.*.sh)|^PROJECT$|^config/*/.*$'
```
[the remark this answers is not in the document]
Comment (m00g3n) [reply prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml]: good point we need to sync module with the image; I've updated the trigger

**Review by kwiatekus — DISMISSED** [review]

File: prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml [hunk 3]
```diff
@@ -167,6 +169,48 @@ presubmits: # runs on PRs
...
+              - "-C"
+              - "hack/ci"
+              - "all"
+            env:
+              - name: IMG
+                value: "europe-docker.pkg.dev/kyma-project/prod/keda-manager:${PULL_BASE_SHA}"
+              - name: KUSTOMIZE_VERSION
+                value: "v4.5.6"
+              - name: MODULE_REGISTRY
+                value: "europe-docker.pkg.dev/kyma-project/dev/unsigned"
```
Comment (kwiatekus) [inline prow/jobs/modules/external/keda-manager/keda-manager-generic.yaml:202]:
```suggestion
                value: "europe-docker.pkg.dev/kyma-project/prod/unsigned"
```

File: templates/data/generic_external_module_data.yaml [hunk 4]
```diff
@@ -54,18 +54,42 @@ templates:
                     - jobConfig_default
                     - jobConfig_presubmit
               - jobConfig:
+                  name: post-keda-module-build
+                  labels:
+                    preset-sa-kyma-push-images: "true"
+                  env:
+                    KUSTOMIZE_VERSION: "v4.5.6"
+                    MODULE_REGISTRY: "europe-docker.pkg.dev/kyma-project/dev/unsigned"
```
Comment (kwiatekus) [inline templates/data/generic_external_module_data.yaml:62]:
```suggestion
                    MODULE_REGISTRY: "europe-docker.pkg.dev/kyma-project/prod/unsigned" 
```

**Review by dekiel — APPROVED** [review]

**Pull request merged**